### PR TITLE
Fix scheduler CPU usage regression

### PR DESCRIPTION
**Pull request by cberner**

### CODE REVIEW

**Comment by dain** [comment]
The commit message says `recently added created a regression`

File: presto-main/src/main/java/com/facebook/presto/execution/scheduler/TopologyAwareNodeSelector.java [hunk 1]
```diff
@@ -173,17 +184,48 @@ public Node selectCurrentNode()
...
+                blocked = firstCompletedFuture(stateChangeFutures, true);
+            }
+            else {
+                blocked = toWhenHasSplitQueueSpaceFuture(blockedExactNodes, existingTasks, calculateLowWatermark(maxPendingForWildcardNetworkAffinity));
+            }
+        }
+        return new SplitPlacementResult(blocked, assignment);
+    }
+
+    private int calculateMaxPendingSplits(int currentDepth, int totalDepth)
```
Comment (martint) [inline presto-main/src/main/java/com/facebook/presto/execution/scheduler/TopologyAwareNodeSelector.java]: Can you explain what this function is trying to compute? Specifically, what's the underlying model it represents? 

For, say, totalDepth = 3, is f(0, 3) + f(1, 3) + f(2, 3) + f(3, 3) supposed to add up to maxPendingSplitsPerTask? From the code, it appears not to be the case, hence my confusion about what it's meant to do.

The comment below `double queueFraction = ...` doesn't help clarify things, unfortunately.

Comment (cberner) [inline presto-main/src/main/java/com/facebook/presto/execution/scheduler/TopologyAwareNodeSelector.java]: It's computing how much of the queue is available to a split with the given network distance. Perhaps renaming `currentDepth` to `splitAffinity` would help?

A split with zero affinity `f(0, 3)` can use only half the queue, whereas one that is local to the worker `f(3, 3)` is allowed to fill the entire queue

Comment (cberner) [inline presto-main/src/main/java/com/facebook/presto/execution/scheduler/TopologyAwareNodeSelector.java]: Yep. Which two sentences?

Comment (martint) [inline presto-main/src/main/java/com/facebook/presto/execution/scheduler/TopologyAwareNodeSelector.java]: "It's computing how much ..." and "a split with zero..."

Comment (cberner) [inline presto-main/src/main/java/com/facebook/presto/execution/scheduler/TopologyAwareNodeSelector.java]: I'll add this:
```
/**
     * Computes how much of the queue can be filled by splits with the network topology distance given by
     * splitAffinity. A split with zero affinity can only fill half the queue, whereas one that matches
     * exactly can fill the entire queue.
     */
```

**Comment by cberner** [comment]
Updated to address comments

**Pull request merged**